install.sh

```diff
@@ -0,0 +1,477 @@
+  fi
+
+  if [[ ! -f "${dir}/CMakeLists.txt" && ! -f "${dir}/.gitmodules" ]]; then
+    [ -n "$VERBOSE" ] && echo "[$SCRIPT_NAME] Cloning $pkg to ${dir}." >&2
+    do_run git clone --recursive ${git_branch_option} "${GIT_BASE_URL}${pkg}.git" "${dir}"
+  else
+    [ -n "$VERBOSE" ] && echo "[$SCRIPT_NAME] Found ${pkg} in ${dir}." >&2
+
+    if [ -e "${dir}/.git" ]; then
+      current_git_branch=`cd "${dir}" && git rev-parse --abbrev-ref HEAD`
+      if [[ -n "$current_git_branch" && -n "$git_branch"
+            && "$current_git_branch" != "$git_branch" ]]; then
+        [ -n "$VERBOSE" ] && echo "[$SCRIPT_NAME] Checking out branch '${git_branch}' of the source repository for package $pkg..." >&2
+        (
+          do_run cd "${dir}" && do_run git checkout ${git_branch}
+        )
+      fi
+    fi
+  fi
+}
+
+function do_build_and_install()
+{
+  # clean build folder
+  if [ "$CLEAN" = true ]; then
+    do_run rm -rf build/
+  fi
+
+  # set CMAKE_BUILD_TYPE
+  if [ -n "$CMAKE_BUILD_TYPE" ]; then
+    CMAKE_ARGS=( -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE} "${CMAKE_ARGS[@]}" )
+  fi
+
+  # set OROCOS_TARGET
+  if [ -n "$OROCOS_TARGET" ]; then
+    CMAKE_ARGS+=( -DOROCOS_TARGET="${OROCOS_TARGET}" )
+  fi
+
+  # add additional cmake options for enabled features
+  FEATURE_CMAKE_ARGS=( )
+  for feature in "${FEATURES[@]}"; do
+    case "$feature" in
+      corba)
+        if is_feature_enabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_CORBA=ON )
+          if [ "${CMAKE_ARGS[$@]}" != "*-DCORBA_IMPLEMENTATION=*" ]; then
+            FEATURE_CMAKE_ARGS+=( -DCORBA_IMPLEMENTATION=${CORBA_IMPLEMENTATION} )
+          fi
+        elif is_feature_disabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_CORBA=OFF )
+        fi ;;
+      mqueue)
+        if is_feature_enabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_MQ=ON )
+        elif is_feature_disabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_MQ=OFF )
+        fi ;;
+      tests)
+        if is_feature_enabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_TESTS=ON )
+        elif is_feature_disabled $feature; then
+          FEATURE_CMAKE_ARGS+=( -DENABLE_TESTS=OFF )
+        fi ;;
+    esac
+  done
+
+  # enable verbosity in make
+  if [ -n "$VERBOSE" ]; then
+    MAKE_ARGS+=( VERBOSE=1 )
+  fi
+
+  # build and install packages
+  for pkg in "${PACKAGES[@]}"; do
+    if [ ! -d "${SOURCES}/$pkg" ]; then continue; fi
+
+    echo "*******************************************************************************" >&2
+    echo "** Building package $pkg" >&2
+    echo "*******************************************************************************" >&2
+    do_run mkdir -p build/$pkg && do_run pushd build/$pkg >/dev/null
+    do_run cmake "${SOURCES}/$pkg" \
+      -DCMAKE_INSTALL_PREFIX="${PREFIX}" \
+      "${FEATURE_CMAKE_ARGS[@]}" "${CMAKE_ARGS[@]}"
+    do_run make ${JOBS} "${MAKE_ARGS[@]}"
+    do_run make install
```

Comment:
Since the install prefix is by default /usr/local, and normal users do not have write rights by default to that folder, it might be ok to add sudo here?